text/0000-jsbox.md

```diff
@@ -0,0 +1,341 @@
+- Feature Name: `JsBox`
+- Start Date: 2020-07-28
+- RFC PR: (leave this empty)
+- Neon Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+`JsBox` is a smart pointer to data created in Rust and managed by the V8 garbage collector. `JsBox` is a basic building block for higher level APIs like neon [classes](https://docs.rs/neon/0.4.0/neon/macro.declare_types.html).
+
+# Motivation
+[motivation]: #motivation
+
+Classes require a substantial amount of boilerplate and place a large amount of code inside of a macro. While feature rich, classes can be unergonomic for general use. In many cases, users only need a way to safely pass references across the FFI boundary.
+
+`JsBox` was [first suggested in 2017](https://github.com/neon-bindings/rfcs/issues/6#issuecomment-353403121). There is also evidence that multiple users of Neon only use classes as a thin wrapper, instead relying on glue code in javascript to create classes by composing methods.
+
+Example of boilerplate required to use a Neon Class as a thin wrapper:
+
+```rust
+pub struct Person {
```

Comment:
Formatting nit: seems strange to use tabs or 8 spaces here; Rust convention is 4-space indentation.

Reply:
It's tabs. I'll switch to spaces because markdown seems to like it better.

With that said, I am a strong supporter of tabs over spaces, regardless of the Rust style guide.

This is an interesting article about why tabs can make your code more accessible and friendly for all developers.
https://dev.to/alexandersandberg/why-we-should-default-to-tabs-instead-of-spaces-for-an-accessible-first-environment-101f